packages/puppeteer-core/src/bidi/core/Browser.ts

```diff
@@ -159,6 +159,9 @@ export class Browser extends EventEmitter<{
   dispose(reason?: string, closed = false): void {
     this.#closed = closed;
     this.#reason = reason;
+    for (const userContext of this.#userContexts.values()) {
```

Comment:
Why don't we just listen for the `close` event on the other object rather then poking directly here?

Reply:
Could you elaborate on what you propose? the browser is disposes, so everything it owns should be disposed. Do you suggest that every object needs to be made aware of the session 'end' event?